iceoryx_hoofs/include/iceoryx_hoofs/cxx/stack.hpp

```diff
@@ -30,9 +32,16 @@ namespace cxx
 /// @tparam T type which the stack contains
 /// @tparam Capacity the capacity of the stack
 template <typename T, uint64_t Capacity>
-class stack // NOLINT (cppcoreguidelines-pro-type-member-init,hicpp-member-init)
+class stack final // NOLINT(cppcoreguidelines-pro-type-member-init, hicpp-member-init)
```

Comment:
In my opinion we should use the `final` keyword. The stack class includes a public d'tor that is not virtual and therefore should not be used as a base class. Once we need the class as a base class, we can remove the `final` keyword and make the d'tor virtual. Since we discussed this topic in other PRs as well, I suggest to find a consensus in a separate issue and, if we decide to disable the rule, create a PR in which we remove all `final` occurrences.

Reply:
We can do this. Its surely is no issue now, but the AUTOSAR rule itself is obsolete in modern C++ with smart pointers.